libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp

```diff
@@ -110,26 +100,23 @@ namespace hpx::execution::experimental::detail {
 
             auto get_value()
             {
-                if (hpx::holds_alternative<value_type>(value))
+                if (hpx::holds_alternative<result_type>(value))
                 {
-                    if constexpr (is_void_result)
-                    {
-                        return;
-                    }
-                    else
-                    {
-                        return HPX_MOVE(hpx::get<value_type>(value));
-                    }
+                    // pull the tuple out of the variant and wrap it into an
+                    // optional
+                    return hpx::make_optional(
+                        hpx::get<0>(hpx::get<result_type>(HPX_MOVE(value))));
```

Comment:
The return issue here: 
`/dev/shm/hpx/src/libs/core/execution/include/hpx/execution/algorithms/sync_wait.hpp:108:36: error: no matching function for call to ‘get<0>(std::variant<>)’
                         hpx::get<0>(hpx::get<result_type>(HPX_MOVE(value))));`

--> I think hpx::make_optional() trys to construct an optional object, but we should get_value() here from optional, so I think we should do 
`return hpx::get<0>(hpx::get<hpx::make_optional(result_type)>(HPX_MOVE(value))));` ?